### DISCUSSION TITLE
Web Service Sender Location URI based on configuration map

### DISCUSSION THREAD

**Discussion by dfauraz** [nextgenhealthcare/connect#5356] [Feature/Enhancement Ideas]
The web service sender Location URI is typically changing between DEV and PROD environments.
However, it is not possible to use a velocity template variable in this field, for example "${my-environment-uri}".
The value must be either empty or hard coded. Hard coded is obviously not an option.
If it is empty, the value is derived from the WSDL URL, assuming the service port address location is set correctly in the WSDL of the different environments.
This may be fine in most cases, but not in all cases:
* WSDL is fix (same in all envs)
* you need to mock the service => this means you also need to mock the WSDL, which is useless
It would be easier just to allow a velocity template variable in Location URI.

**Comment by siddharthsharma9258** [comment]
`configurationMap.get('URI')` does not work?

**Reply by dfauraz** [reply]
No : you are mixing velocity and Javascript context

**Reply by dfauraz** [reply]
Look at other connectors, like the TCP connector: host and port can be set to a certain hard coded value or to "${my-host}" or "${my-port}". The results is that Mirth Connect is going to look for the given variables in the context maps (including the conf map) and derive host and port depending on the environment. Here I suggest to have the same thing for SOAP web service Location URI

**Reply by pacmano1** [reply]
@dfauraz , agreed would be nice - did you add it to the issues/features section?    I don't use the web service sender anymore though, rathar an http sender for SOAP calls.
